### DISCUSSION TITLE
Gitea - autodiscover by organizations

### DISCUSSION THREAD

**Discussion by ronaldvb-k** [renovatebot/renovate#26792] [Suggest an Idea]
### Tell us more.

We are using `autodiscover` and `autodiscoverFilter` to filter out repositories based on an Gitea organizations. Renovate retrieves all repositories that the bot account can access. On our Gitea hosted instance, we have thousands of projects. It takes a long time to retrieve all these projects and it adds an additional unwanted load on our Gitea instance.

We would like to retrieve repositories only from a list of Gitea organizations.

This has already been implemented for Gitlab (https://github.com/renovatebot/renovate/issues/12635). And it would be nice if it also can be implemented for Gitea.

Gitea has an endpoint to retrieve repositories based on a organization name: https://try.gitea.io/api/swagger#/organization/orgListRepos. The API does not support filtering out achived repositories, so we have to filter them out by ourself.

**Comment by viceice** [comment]
you can probably workaround by using the new topic filter

https://github.com/renovatebot/renovate/releases/tag/37.141.0

**Reply by ronaldvb-k** [reply]
Sounds like good workaround, so i tested the new topic filter on our test environment.
It makes searching faster, so that's good!

Unfortunately, it's not possible to set topics on organization level inside Gitea. And as each teams are managing their own repositories inside an organization, it's not possible for us to create a topic for all repositories. Teams can create their own repository, and will probably forget to set the right topic on a repository.
